Modules/socketmodule.c

```diff
@@ -6588,28 +6590,54 @@ Set the default timeout in seconds (float) for new socket objects.\n\
 A value of None indicates that new socket objects have no timeout.\n\
 When the socket module is first imported, the default is None.");
 
-#ifdef HAVE_IF_NAMEINDEX
+#if defined(HAVE_IF_NAMEINDEX) || defined(MS_WINDOWS)
 /* Python API for getting interface indices and names */
 
 static PyObject *
 socket_if_nameindex(PyObject *self, PyObject *arg)
 {
-    PyObject *list;
+    PyObject *list = PyList_New(0);
+    if (list == NULL) {
+        return NULL;
+    }
+#ifdef MS_WINDOWS
+    PMIB_IF_TABLE2 tbl;
+    if (GetIfTable2Ex(MibIfTableRaw, &tbl) != NO_ERROR) {
+        Py_DECREF(list);
+        return PyErr_SetFromErrno(PyExc_OSError);
```

Comment:
I think we want `SetFromWindowsError` here (or something like that). `SetFromErrno` will use the POSIX `errno` variable, while we want to use `GetLastError()`.

Reply:
Apparently, `GetLastError()` can't be used with [GetIfTable2Ex()](https://docs.microsoft.com/en-us/windows/desktop/api/netioapi/nf-netioapi-getiftable2ex#return-value) or [ConvertInterfaceLuidToNameW()](https://docs.microsoft.com/en-us/windows/desktop/api/netioapi/nf-netioapi-convertinterfaceluidtonamew#return-value). The return value of those two functions must be used (I've verified this locally).